### PR TITLE
CB2-1728 add retry logic to database call

### PR DESCRIPTION
**Pull request by Ian-Hodges**
Changes made so the retry policy of eventbridge can be used to handle the retry logic.

### CODE REVIEW

**Comment by Ian-Hodges** [comment]
## Tested in AWS.

### Test setup 1:
1. Eventbridge set to retry 3 times.
2. Lambda configure to error.

#### Result:
Lambda called three times as shown in the lambda logs.
image

### Test setup 2:
1. Eventbridge set to retry 3 times.
2. Lambda configure to succeed.

#### Result:
Lambda called once as shown in the lambda logs.
image

File: src/handler.ts [hunk 1]
```diff
@@ -28,12 +28,11 @@ const handler = async (event: ScheduledEvent): Promise<{ statusCode: number; bod
     await sendEvents(facilitySchedules);
 
     logger.info('Data processed successfully.');
-    return { statusCode: 200, body: 'Data processed successfully.' };
+    callback(null, 'Data processed successfully.');
```
Comment (aaron-downing) [inline src/handler.ts:31]: does the callback provide a 200 status when successful?

Comment (Ian-Hodges) [reply src/handler.ts:31]: It doesn't. I'm not sure how eventbridge calls a lambda, but the statusCode is not needed.

**Review by aaron-downing — APPROVED** [review]

**Review by tom-bjss — APPROVED** [review]

File: src/handler.ts (hunk 1 above)
Comment (tom-bjss) [reply src/handler.ts:31]: Will this mess with manual invocations with regards to response codes ?

Comment (Ian-Hodges) [reply src/handler.ts:31]: No. This change has no change to the response codes being returned.

**Pull request merged**